File: 331. Verify Preorder Serialization of a Binary Tree.py
class Solution(object):
    def isValidSerialization(self, preorder):
        """
        :type preorder: str
        :rtype: bool
        """
        nodes = preorder.split(',')
        diff = 1
        for x in nodes:
            diff -= 1
            if diff < 0:
                return False
            if x != '#':
                diff += 2
        return diff == 0

File: go/src/leetcode/331. Verify Preorder Serialization of a Binary Tree/331. Verify Preorder Serialization of a Binary Tree.go
package main

import "strings"

func isValidSerialization(preorder string) bool {
	diff := 1
	nodes := strings.Split(preorder, ",")
	for _, node := range nodes {
		diff--
		if diff < 0 {
			return false
		}
		if node != "#" {
			diff += 2
		}
	}
	return diff==0
}
